transport/grpc/outbound_test.go

```diff
@@ -200,3 +202,84 @@ func TestCallStreamWithInvalidPeer(t *testing.T) {
 		}.Error(),
 	)
 }
+
+func TestCallSvcMatch(t *testing.T) {
+	tests := []struct {
+		msg         string
+		serverPort  string
+		headerKey   string
+		headerValue string
+		wantErr     bool
+	}{
+		{
+			msg:         "call service match success",
+			serverPort:  "15213",
+			headerKey:   ServiceHeader,
+			headerValue: "Service",
+		},
+		{
+			msg:         "call service match failed",
+			serverPort:  "15619",
+			headerKey:   ServiceHeader,
+			headerValue: "ThisIsWrongSvcName",
+			wantErr:     true,
+		},
+		{
+			msg:        "no service name response header",
+			serverPort: "15640",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.msg, func(t *testing.T) {
+			server := grpc.NewServer(
+				grpc.CustomCodec(customCodec{}),
```

Comment:
I suspect you don't need this because you're not implementing "real" handlers
here. If it works without this option, I would prefer to drop it so that this
test doesn't rely on implementation details of how we integrate with gRPC. The
functionality being tested here theoretically only needs the plain gRPC APIs
for the server.

Reply:
I got panic message about encoding error when I remove that line, so this line maybe necessary.